extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/Basic.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.vertx.http.security;
+
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+/**
+ * Represents a Basic authentication configuration.
+ */
+public record Basic(String realm, Optional<Boolean> enabled) {
```

Comment:
I have to admit, I'm not sure right now. I mean, this looks neat, and well thought out, I'm just not sure it works for a generic HttpSecurity level fluent API, as we can't offer the same for other mechanisms. I'll have to think also try to suggest some alternatives and we should invite more reviewers.

Reply:
OK, as long as you have this on mind, take your time. I'd prefer if this got in for 3.25, but it's not a big deal and if so, there is still some time.